src/redux-middleware/pushQueue.ts

```diff
@@ -108,27 +108,42 @@ const flushEdits = (pushQueue: PushBatch[]): Thunk<Promise<void>> => async (disp
 /** Pull all descendants of pending moves and dispatch existingThoughtMove to fully move. */
 const flushMoves = (pushQueue: PushBatch[]): Thunk => async (dispatch, getState) => {
 
-  // if there are pending thoughts that need to be moved, dispatch an action to be picked up by the pullQueue middleware which can load pending thoughts before dispatching another existingThoughtMove
-  const pendingMoves = pushQueue.map(batch => batch.pendingMoves || []).flat()
-  if (pendingMoves?.length) {
+  // if there are pending thoughts that need to be deleted, dispatch an action to be picked up by the pullQueue middleware which can load pending thoughts before dispatching another existingThoughtDelete
+  const descendantMoves = pushQueue.map(batch => batch.descendantMoves || []).flat()
+  const pendingPulls = pushQueue.map(batch => batch.pendingPulls || []).flat()
+
+  let maxDepth = Infinity
+  if (descendantMoves?.length) {
 
-    const pending: Index<Context> = keyValueBy(pendingMoves, ({ pathOld }) => {
+    const pending: Index<Context> = keyValueBy(descendantMoves, ({ pathOld }) => {
       const context = pathToContext(pathOld)
-      return {
-        [hashContext(context)]: context
-      }
+      // skip the pull for loaded descendants
+      return { [hashContext(context)]: context }
     })
 
     await dispatch(pull(pending, { maxDepth: Infinity }))
+    maxDepth = getDepth(getState(), pathToContext(descendantMoves[0].pathOld))
```

Comment:
`descendantMoves` combines multiple batches (L112), so `descendantMoves[0]` is not necessarily the root of all `descendantMoves`. For example, if there was a batch that involved moving thoughts with depth 3 and a batch that involved moving thoughts with depth 10, this would mistakenly load only 3 levels.

It would be more correct to use a different pull depth for each batch, but we don't want the overhead of separate pulls. What do you think? We could call `getDepth` on the root of each batch and take the max. That would cover all the cases, and the the inefficiency of pulling a few of the batches too deep is better than the inefficiency of multiple pulls or the data integrity issues of not pulling a batch deeply enough.